apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex

```diff
@@ -194,6 +307,86 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
     {:noreply, assign(socket, current_data_dictionary_item: field, selected_field_id: field_id)}
   end
 
+  defp generate_schema(decoded_file, dataset_id) do
+    decoded_file
+    |> SchemaGenerator.generate_schema()
+    |> Enum.map(&assign_schema_field_details(&1, dataset_id, nil))
+  end
+
+  defp assign_schema_field_details(schema_field, dataset_id, parent_bread_crumb) do
```

Comment:
I wonder if some of this couldn't go in a helper of some kind, so it can be tested more directly?